plugins/usbdmx/DMXCreator.cpp

```diff
@@ -0,0 +1,257 @@
+  memset(tmp_buffer, 0, length);
+  buffer.Get(tmp_buffer, &length);
+
+  if (memcmp(tmp_buffer, m_control_setup_buffer, CHANNELS_PER_PACKET) == 0) {
+    // no change -> don't send anything
+    return true;
+  }
+
+  unsigned char status_buffer[6] = {
+    0x80, 0x01, 0x00, 0x00, 0x00, 0x01
+  };
+
+  int bytes_sent = 0;
+
+  length = CHANNELS_PER_PACKET;
+  memset(m_control_setup_buffer, 0, length);
+  buffer.Get(m_control_setup_buffer, &length);
+
+  int r = m_adaptor->BulkTransfer(handle, ENDPOINT_1, status_buffer,
+    sizeof(status_buffer), &bytes_sent, URB_TIMEOUT_MS_SYNC);
+
+  OLA_INFO << "Sending status bytes returned " << r;
+
+  // Sometimes we get PIPE errors, those are non-fatal
+  if (r < 0 && r != LIBUSB_ERROR_PIPE) {
+    OLA_WARN << "Sending status bytes failed";
+    return false;
+  }
+
+  bytes_sent = 0;
+  r = m_adaptor->BulkTransfer(handle, ENDPOINT_2,
+    const_cast<unsigned char*>(m_control_setup_buffer),
+    CHANNELS_PER_PACKET, &bytes_sent, URB_TIMEOUT_MS_SYNC);
+  OLA_INFO << "Sending data bytes returned " << r;
+  return r >= 0 || r == LIBUSB_ERROR_PIPE;
+}
+
+
+// SynchronousDMXCreator
+// -----------------------------------------------------------------------------
+
+SynchronousDMXCreator::SynchronousDMXCreator(LibUsbAdaptor *adaptor,
+                                             libusb_device *usb_device)
+    : DMXCreator(adaptor, usb_device) {
+}
+
+bool SynchronousDMXCreator::Init() {
+  libusb_device_handle *usb_handle;
+
+  bool ok = m_adaptor->OpenDeviceAndClaimInterface(
+      m_usb_device, 0, &usb_handle);
+  if (!ok) {
+    return false;
+  }
+
+  std::auto_ptr<DMXCreatorThreadedSender> sender(
+      new DMXCreatorThreadedSender(m_adaptor, m_usb_device, usb_handle));
+  if (!sender->Start()) {
+    return false;
+  }
+  m_sender.reset(sender.release());
+  return true;
+}
+
+bool SynchronousDMXCreator::SendDMX(const DmxBuffer &buffer) {
+  return m_sender.get() ? m_sender->SendDMX(buffer) : false;
+}
+
+// DMXCreatorAsyncUsbSender
+// -----------------------------------------------------------------------------
+class DMXCreatorAsyncUsbSender : public AsyncUsbSender {
+ public:
+  DMXCreatorAsyncUsbSender(LibUsbAdaptor *adaptor, libusb_device *usb_device)
+      : AsyncUsbSender(adaptor, usb_device),
+        m_adaptor(adaptor),
+        m_usb_device(usb_device) {
+    m_dmx_buffer_1 = new uint8_t[CHANNELS_PER_PACKET];
+    m_dmx_buffer_2 = new uint8_t[CHANNELS_PER_PACKET];
+    m_state = STATE_SEND_STATUS;
+  }
+
+  ~DMXCreatorAsyncUsbSender() {
+    CancelTransfer();
+    delete[] m_dmx_buffer_1;
+    delete[] m_dmx_buffer_2;
+  }
+
+  libusb_device_handle* SetupHandle() {
+    libusb_device_handle *usb_handle;
+    bool ok = m_adaptor->OpenDeviceAndClaimInterface(
+        m_usb_device, 0, &usb_handle);
+    return ok ? usb_handle : NULL;
+  }
+
+  bool PerformTransfer(const DmxBuffer &buffer) {
+    // if we only wanted to send the first half, the last byte would be 0x01
```

Comment:
Unfortunately not. I just have this one device for testing, I guess the other DMXCreator dongles use other bytes.